src/uct/ib/base/ib_md.c

```diff
@@ -315,13 +329,13 @@ void *uct_ib_md_mem_handle_thread_func(void *arg)
               (ctx->params != NULL) ? "reg_mr" : "dereg_mr",
               start, ctx->address, ctx->first_mr_size,
               ucs_time_to_usec(ucs_get_time() - t0));
-    return UCS_STATUS_PTR(UCS_OK);
+    return (void*)(intptr_t)mr_idx;
```

Comment:
returning either count or status seems error-prone
Maybe thread can just update it's `mr_count` inside?
And always return status

Reply:
I dont know how is it omportant here, but usually we use ssize_t for such purpose, can see bcopy API

Reply:
this is unrelate to my PR, bad rebase, pushed a revert commit @evgeny-leksikov @iyastreb